app/controllers/tasks_controller.rb

```diff
@@ -53,8 +54,14 @@ def task_params
     params.require(:task).permit(:status, :title).tap do |task|
       if task[:status].present?
         task[:status] = Task.statuses[task["status"]]
-        task[:finished_by] = current_user.id
         task[:week] = find_week.number
+        if (task[:status] == 2) 
+              task[:finished_by] = current_user.id
+              task[:completed_at] = DateTime.now
+         else  
+              task[:finished_by] = nil
+              task[:completed_at] = nil
+        end
```

Comment:
I have done this because, if someone unticks the task(marks uncomplete), then these fields were not getting nil.

Reply:
👍